signer/src/storage/model.rs

```diff
@@ -1352,6 +1346,256 @@ pub struct WithdrawalRejectEvent {
+    }
+}
+impl From<u32> for StacksBlockHeight {
+    fn from(value: u32) -> Self {
+        Self(value as u64)
+    }
+}
+impl From<u64> for StacksBlockHeight {
+    fn from(value: u64) -> Self {
+        Self(value)
+    }
+}
+impl From<usize> for StacksBlockHeight {
+    fn from(value: usize) -> Self {
+        Self(value as u64)
+    }
+}
+
+// Conversion StacksBlockHeight => u64  is not implemented intentionally.
+// Use deref instead.
+// This was done for consistency across the codebase.
+
+impl From<StacksBlockHeight> for u128 {
+    fn from(value: StacksBlockHeight) -> Self {
+        *value as u128
+    }
+}
+
+impl std::fmt::Display for StacksBlockHeight {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+impl Deref for StacksBlockHeight {
+    type Target = u64;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+impl TryFrom<StacksBlockHeight> for i64 {
+    type Error = TryFromIntError;
+    fn try_from(value: StacksBlockHeight) -> Result<Self, Self::Error> {
+        i64::try_from(value.0)
+    }
+}
+
+impl TryFrom<i64> for StacksBlockHeight {
+    type Error = TryFromIntError;
+    fn try_from(value: i64) -> Result<Self, Self::Error> {
+        u64::try_from(value).map(Self)
+    }
+}
+
+impl Add<u64> for StacksBlockHeight {
+    type Output = StacksBlockHeight;
+    fn add(self, other: u64) -> Self::Output {
+        Self(self.0.add(other))
+    }
+}
+impl Add<StacksBlockHeight> for u64 {
+    type Output = StacksBlockHeight;
+    fn add(self, other: StacksBlockHeight) -> Self::Output {
+        StacksBlockHeight((self).add(other.0))
+    }
+}
+impl Add for StacksBlockHeight {
+    type Output = StacksBlockHeight;
+    fn add(self, other: StacksBlockHeight) -> Self::Output {
+        Self(self.0.add(other.0))
+    }
+}
+
+impl Sub<u64> for StacksBlockHeight {
+    // Height - int is still height.
+    type Output = StacksBlockHeight;
+    fn sub(self, other: u64) -> Self::Output {
+        StacksBlockHeight((*self).sub(other))
+    }
+}
+impl Sub for StacksBlockHeight {
+    // Diff of two heights is int, not height.
+    type Output = u64;
+    fn sub(self, other: StacksBlockHeight) -> Self::Output {
+        self.0.sub(other.0)
+    }
+}
+impl StacksBlockHeight {
+    /// Behaves same as u64.saturating_add
+    pub fn saturating_add(self, rhs: impl Into<StacksBlockHeight>) -> Self {
+        let rhs: u64 = rhs.into().0;
+        Self(self.0.saturating_add(rhs))
+    }
+
+    /// Behaves same as u64.saturating_sub
+    pub fn saturating_sub(self, rhs: impl Into<StacksBlockHeight>) -> Self {
+        let rhs: u64 = rhs.into().0;
+        Self(self.0.saturating_sub(rhs))
+    }
+}
+
+/// Bitcoin block height
+#[derive(
+    Debug,
+    Default,
+    Clone,
+    Copy,
+    Hash,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Serialize,
+    Deserialize,
+    sqlx::FromRow,
+)]
+pub struct BitcoinBlockHeight(u64);
```

Comment:
I don't think `sqlx::FromRow` is needed for these, but maybe I'm mistaken. Also, do we need `Deserialize` or `Default`?

Reply:
And we probably want `#[serde(transparent)]` on these as well.

Reply:
Added transparent here: [2c70343](https://github.com/stacks-sbtc/sbtc/pull/1448/commits/2c703435c93dba2f20d67bd79f290213b4a6b3c3)

Reply:
About `Deserialize` and `Default` I'd say even if we don't need them now there is no problem to have them. They don't make code more difficult to understand since this are the most basic `derive's` rust have and this list of `derive's` will be still huge even if we remove them. Also they do not slow down code execution. The only drawback they can bring -- they proooobably will make binary slightly bigger in some builds (but I think compiler will eliminate dead code in release builds), but even for debug builds, binary size is not really important for us I think

Reply:
About `FromRow`: removed it here: [f5b6030](https://github.com/stacks-sbtc/sbtc/pull/1448/commits/f5b60304edb7a923b103eadb35b0648b802cde71). Compared to `Deserialize` and `Default` it is much less obvious what this `derive` does imo, so it make sense to remove it

Reply:
I usually just derive what is needed, because I cannot think of a good reason to derive something if it isn't needed. Same thing for code that is unused. Well, at least that's my rule of thumb. You can probably find some derived impls that are unused or were never used. We should probably remove them.

For `Default`, I don't think that we should have one, except maybe in tests. My thinking is that it can be a footgun for production code. For `Deserialize`, we don't implement it for most types in this crate, including the majority in this module (and I wonder if the ones in this module can be removed). It's small code for each one but it would add up to a lot if we did it for all of our types. Also, if it's unused, why add it.

Reply:
Whoops, actually, `Deserialize` is required by `SignerConfig` and default by 
```
impl<const N: usize> ReorgDescription<N> {
    fn num_blocks(&self) -> u64 {
        *self.sweep_heights.into_iter().max().unwrap_or_default()
    }
}
```

Reply:
Fair enough!